src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -869,6 +864,36 @@ private SafeCurlSlistHandle SetRequestHeaders(SafeCurlHandle handle, HttpRequest
             return retVal;
         }
 
+        /// <summary>
+        /// Prepare headers in a format that can be fed to curl API
+        /// </summary>
+        /// <param name="headers"></param>
+        /// <returns>List of headers</returns>
+        private static List<string> PrepareRequestHeaders(params HttpHeaders[] headers)
+        {
+            List<string> allHeaders = new List<string>(headers.Length);
+
+            foreach (var multiHeaders in headers)
+            {
+                if (multiHeaders != null)
+                {
+                    foreach (var header in multiHeaders)
+                    {
+                        string headerStr = header.Key + ":";
+
+                        foreach (var headerValue in header.Value)
+                        {
+                            headerStr += " " + headerValue;
```

Comment:
Mutliple headers with same name, if used causes issue. The response is Bad Request. WinHttp also does not send multiple headers while sending out a multi value header. It merges them either by space or by comma.
I have now updated the code to use the same method to generate header values that is used by winHttpHandler. The method uses comma or space as delimitter based on some logic. libcurl does not do anything extra with space separated, it passes on the header as it is.